### ISSUE TITLE
Major changes should be in branch until supported by all clients

### ISSUE THREAD

**Issue by yurivict** [irungentoo/toxcore#1326]
Currently qTox is broken due to major changes in toxcore: https://github.com/tux3/qTox/issues/1505 

You shouldn't roll out major changes before making sure major clients are compatible.
What users why are adopting tox are supposed to do?

**Comment by LittleVulpix** [comment]
I kind of concur because nightly/linux users already have the 0.2.p and they are causing wild crazy chat incompatibilities + random weird file being sent (due to the incompatibilities between old api and new api) when they come online for someone who is using 0.2.o (latest windows "weekly").

I guess the most problematic part is people who just rely on the autoupdater without being technically skilled. There are more than just "developers" using tox now. Grandmas and moms, and they will suddenly have big trouble due to this.

tl;dr - switch over instantly all at once, when everything is ready; not one by one, because this is causing a lot of confusion / many issues.

I understand the wait is because qtox currently doesn't have an available dev (i.e. no time due to random circumstances), but I'd prefer to not be in this "state of weirdness" for very long.

**Comment by yurivict** [comment]
I would roll back toxcore to the working state and wait for clients to catch up if I was the toxcore project owner.

**Comment by irungentoo** [comment]
The new api was merged to master because it was ready to be merged.

All our recommended binaries are still on the old api because we are waiting for all clients (qTox) to work on it before pushing an update.

**Comment by yurivict** [comment]
I see, people just don't know about api_old_version branch! (You should tell downstreams when such major event occurs, because ex. Arch just picks up the top of github master in their AUR package)

I fixed FreeBSD port. Arch is still messed up. deb/rpm/etc -- who knows ...

**Comment by fcore117** [comment]
Everyone who are in my contact list has upgraded to newapi manually.

**Comment by yurivict** [comment]
@fcore117  @irungentoo  Toxic doesn't compile with the old_api release, see here https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=199362

There *must* be some version set for toxcore and every major client that all work together.

**Comment by irungentoo** [comment]
old api toxcore: https://github.com/irungentoo/toxcore/commit/4ad76497881ee2a623acdedcf0ac10406208b716

This is the last version of toxic that works on the old api: https://github.com/Tox/toxic/commit/c34ecc8dfde5764294a0cbbdb4212f708a21838f

This is the last version of uTox that works on the old api: https://github.com/notsecure/uTox/commit/bdd1a009518e325a73a527da152793102b765ef6

Latest qTox master is still on the old api.

Latest uTox and Toxic master are new api and work with latest toxcore master.

**Comment by yurivict** [comment]
@irungentoo  No, we need to have on toxcore version working with *all* clients.
So it can't be "this one with old API, this one with new API".

We need one toxcore (I presume it is old_api branch because of qTox), and the version of each client compatible with it.

**Comment by irungentoo** [comment]
@yurivict then use the old api, I gave you the last commits for toxic and uTox that work with old api.

**Comment by suhr** [comment]
The old api was too broken to keep any support of it. Lack of the new api support in qTox is a qTox issue, not a toxcore one.

**Comment by yurivict** [comment]
So what are users supposed to do?

Arch users are stuck withe broken qTox. BSD users also stuck with the broken qTox.
qTox is the only client that supports voice/audio. Explanation that "it's not our problem" doesn't work Сухарик.

**Comment by Impyy** [comment]
@yurivict When did uTox stop supporting audio?

**Comment by yurivict** [comment]
When I originally installed qTox last year, it was the only one. But this isn't the point. qTox users  need to have some working solution.

We in BSD have the packaging system. There is one package for toxcore, and there are many packages for other clients. Same with Arch actually, I checked 3 days ago. I am sure it is the same for rpm systems. How are all thee users supposed to continue using Tox?

**Comment by suhr** [comment]
qTox must be moved to the new api. Too bad there's a few volunteers to do it.
See https://github.com/tux3/qTox/issues/1520

**Comment by landswellsong** [comment]
I've volounteered will probably ship tomorrow provided a good wind.

**Comment by yurivict** [comment]
Thank you!

**Comment by zetok** [comment]
@yurivict https://lists.tox.im/pipermail/general/2015-April/000028.html here you go ``:)``

**Comment by yurivict** [comment]
Thanks!

**Comment by GrayHatter** [comment]
Shouldn't happen again (crosses fingers for tomorow's new API)

But either way, this is not a toxcore issue, tell your client dev's to update.
